src/api/user/user.service.ts

```diff
@@ -0,0 +1,62 @@
+import jwt from 'jsonwebtoken'
+import HttpException from '../../utils/httpException'
+import userModel, { UserDocument } from './user.model'
+import { UserRequest } from './user.type'
+import userUtil from './user.util'
+
+export default class UserService {
+  getAllUser() {
+    return userModel.find()
+  }
+  getUserByEmail(email: string) {
+    return userModel.findOne({ email })
+  }
+
+  // create user
+  createUser(user: UserRequest) {
+    return new Promise<UserDocument>(async (resolve, reject) => {
+      try {
+        const existingUser = await this.getUserByEmail(user.email)
+        if (existingUser) throw new HttpException(401, 'email already exist')
+
+        const hash = await userUtil.hash(user.password)
+        userModel.init()
+        const userDocument = await userModel.create({
+          email: user.email,
+          password: hash,
+        })
+        resolve(userDocument)
+      } catch (error) {
+        reject(error)
+      }
+    })
+  }
+
+  // auth user
+  authenticateUser(user: UserRequest) {
+    return new Promise<string>(async (resolve, reject) => {
+      try {
+        const userDocument = await userModel.findOne({
+          email: user.email,
+        })
+        if (!userDocument)
+          throw new HttpException(401, 'email or password wrong')
+
+        const isPasswordMatch = await userUtil.compare(
+          user.password,
+          userDocument.password
+        )
+        if (!isPasswordMatch)
+          throw new HttpException(401, 'email or password wrong')
```

Comment:
mungkin if nya lebih simple bisa dijadikan 1 saja karena sama sama `throw new HttpException(401, 'email or password wrong')`

Reply:
yang if(!userDocument) itu aku jadikan buat throw error "email wrong"
terus yang if(!isPasswordMatch) itu throw error "password wrong"

Reply:
jadi aku bikin if nya tetep sendiri sendiri. kira - kira gimana kalo kyk gitu?

Reply:
kalo aku coba jadi in satu pake if (!userDocument && !isPasswordMatch),
jadinya userDocument.password yang buat ngecek password nya error